tests/ui/methods.rs

```diff
@@ -443,3 +443,17 @@ fn main() {
     let opt = Some(0);
     let _ = opt.unwrap();
 }
+
+/// Checks implementation of `UNNECESSARY_FILTER_MAP` lint
```

Comment:
It would be nice to move these tests to their own `tests/ui/unnecessary_filter_map.rs` file so that the UI diffs are smaller. (#2038)